src/python/pants/backend/python/target_types.py

```diff
@@ -381,6 +381,24 @@ class PexArgsField(StringSequenceField):
     )
 
 
+class PexCheckField(StringField):
+    alias = "check"
+    valid_choices = ("none", "warn", "error")
+    expected_type = str
+    default = "error"
```

Comment:
I think it'd be good to keep this at `warn`, to avoid breaking existing users who have a build that works for them when they upgrade pants (e.g. maybe they manipulate the resulting zipapp as a zip file directly somehow... potentially better served with a different layout, but we shouldn't be encouraging that with a hard-error). It'd definitely be fine to switch the default with some sort of deprecation cycle. 

```suggestion
    default = "warn"
```

Of course, pants swallowing warnings from PEX isn't so good, but I see you have #20480 to help.

Reply:
I understand the judgement call going either way.  Is there an example for what the deprecation system looks like for changing a default?  I saw lots of examples for removing fields/flags, but not changing a default.

Reply:
I can't think of a specific example exactly like this, but there's occasionally options in `pants.toml` to control global default behaviour, e.g. https://www.pantsbuild.org/2.13/reference/global-options#use_deprecated_directory_cli_args_semantics, so I could imagine putting an option in the `[pex]`  subsystem or similar.

(Unresolving just to continue this conversation.)